packages/fluentui/react-northstar/src/components/Video/Video.tsx

```diff
@@ -24,77 +28,86 @@ export interface VideoProps extends UIComponentProps {
 
   /** Video source URL. */
   src?: string;
+
+  accessibility?: Accessibility<VideoBehaviorProps>;
 }
 
 export const videoClassName = 'ui-video';
 
-class Video extends UIComponent<WithAsProp<VideoProps>> {
-  static create: ShorthandFactory<VideoProps>;
-
-  static deprecated_className = videoClassName;
-
-  static displayName = 'Video';
-
-  static propTypes = {
-    ...commonPropTypes.createCommon({
-      children: false,
-      content: false,
-    }),
-    autoPlay: PropTypes.bool,
-    controls: PropTypes.bool,
-    loop: PropTypes.bool,
-    muted: PropTypes.bool,
-    poster: PropTypes.string,
-    src: PropTypes.string,
-  };
-
-  static defaultProps = {
-    as: 'video',
-    controls: true,
-    autoPlay: false,
-    muted: false,
-  };
-
-  videoRef = React.createRef<HTMLVideoElement>();
-
-  componentDidMount() {
-    this.setVideoAttributes();
-  }
-
-  componentDidUpdate() {
-    this.setVideoAttributes();
-  }
-
-  setVideoAttributes = () => {
+export type VideoStylesProps = Required<Pick<VideoProps, 'variables'>>;
+
+export const Video: React.FC<WithAsProp<VideoProps>> & FluentComponentStaticProps<VideoProps> = props => {
+  const context: ProviderContextPrepared = React.useContext(ThemeContext);
+  const { setStart, setEnd } = useTelemetry(Video.displayName, context.telemetry);
+  setStart();
+  const { controls, autoPlay, loop, poster, src, muted, variables, className } = props;
+  const ElementType = getElementType(props);
+  const videoRef = React.createRef<HTMLVideoElement>();
+  const unhandledProps = useUnhandledProps(Video.handledProps, props);
+  const getA11yProps = useAccessibility(props.accessibility, {
+    debugName: Video.displayName,
+  });
+
+  React.useEffect(() => {
     // React doesn't guaranty that props will be set:
     // https://github.com/facebook/react/issues/10389
-    if (this.videoRef.current) {
-      this.videoRef.current.muted = !!this.props.muted;
+    if (videoRef.current) {
+      videoRef.current.muted = !!muted;
     }
-  };
-
-  renderComponent({ accessibility, ElementType, classes, unhandledProps }) {
-    const { controls, autoPlay, loop, poster, src } = this.props;
-
-    return (
-      <Ref innerRef={this.videoRef}>
-        <ElementType
-          autoPlay={autoPlay}
-          className={classes.root}
-          controls={controls}
-          loop={loop}
-          poster={poster}
-          src={src}
-          {...accessibility.attributes.root}
-          {...unhandledProps}
-        />
-      </Ref>
-    );
-  }
-}
+  }, [muted]);
+
+  const { classes, styles } = useStyles<VideoStylesProps>(Video.displayName, {
+    className: videoClassName,
+    mapPropsToInlineStyles: () => ({ className, variables }),
+  });
+
+  const element = (
+    <Ref innerRef={videoRef}>
+      <ElementType
+        {...getA11yProps('root', {
+          className: classes.root,
+          styles,
```

Comment:
```suggestion
```

There is no need to pass `styles` to a HTML element, it will produce React warning

Reply:
Done